pkg/validation/test_test.go

```diff
@@ -990,6 +990,44 @@ func TestValidateDependencies(t *testing.T) {
 	}
 }
 
+func TestValidateDNSConfig(t *testing.T) {
+	var testCases = []struct {
+		name   string
+		input  []api.StepDNSConfig
+		output []error
+	}{
+		{
+			name:  "no searches",
+			input: nil,
+		},
+		{
+			name: "valid searches",
+			input: []api.StepDNSConfig{
+				{Searches: []string{"search1", "search2"}},
+			},
+		},
+		{
+			name: "invalid searches",
+			input: []api.StepDNSConfig{
+				{Searches: []string{"", ""}},
+			},
+			output: []error{
+				errors.New("root.searches[0] must be set"),
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			if actual, expected := validateDNSConfig("root", testCase.input), testCase.output; !reflect.DeepEqual(actual, expected) {
```

Comment:
You can avoid `reflect.DeepEqual` by only using `cmp.Diff`, resulting in simpler code:

```
if diff := cmp.Diff(one, two, comparer); diff != "" {
t.Errorf("...%s", diff)
}
```